ReplayKitExample/ReplayKitExample/ReplayKitVideoSource.swift

```diff
@@ -24,17 +24,33 @@ class ReplayKitVideoSource: NSObject, VideoSource {
     }
 
     // In order to save memory, the handler may request that the source downscale its output.
-    static let kDownScaledMaxWidthOrHeight = UInt(886)
-    static let kDownScaledMaxWidthOrHeightSimulcast = UInt(1280)
-
-    // Maximum bitrate (in kbps) used to send video.
+    static let kDownScaledMaxWidthOrHeight = UInt(1280)
+    // 1152x531.
+    // static let kDownScaledMaxWidthOrHeightScreencastApp = UInt(1152)
+    static let kDownScaledMaxWidthOrHeightScreencastApp = UInt(1280)
+    // 768x432: 28 - crash MB. 33% CPU
+    // 768x354: 30 MB (s), 48% CPU, 1000 Kbps
+    // iPhone 16:9
+    static let kDownScaledMaxWidthOrHeightScreencastExtension = UInt(768)
+    static let kDownScaledMaxWidthOrHeightScreencastExtensioniPad = UInt(1024)
+
+    static let kDownScaledMaxWidthOrHeightVideoExtension = UInt(768)
+    static let kDownScaledMaxWidthOrHeightVideoApp = UInt(800)
+    static let kDownScaledMaxWidthOrHeightSimulcastApp = UInt(1280)
+    // 832x624: 35 - 41 MB (!!).
+    // 832x468: 30 - 35 - crash MB.
+    // static let kDownScaledMaxWidthOrHeightSimulcast = UInt(832)
+
+    // Maximum bitrate (in kbps) used to send unicast video.
     static let kMaxVideoBitrate = UInt(1440)
-    // The simulcast encoder allocates bits for each layer.
-    static let kMaxVideoBitrateSimulcast = UInt(1180)
+    // Bitrate hint (in kbps) for the simulcast encode.
+    static let kMaxVideoBitrateSimulcast = UInt(1000)
     static let kMaxScreenshareBitrate = UInt(1600)
 
     // Maximum frame rate to send video at.
     static let kMaxVideoFrameRate = UInt(15)
+    static let kMaxScreencastFrameRateApp = UInt(30)
+    static let kMaxScreencastFrameRate = UInt(15)
```

Comment:
I feel we have too many settings here for a sample app. Also sharing the ReplayKitVideoSource between the extension and the app makes it even more complicated.

 Is it possible to pass a configuration object from app and the extension?

Reply:
I like this idea because it is really confusing to have the settings in multiple places. Would be nice to tie it together with just one config object.